src/MinusFilter.java

```diff
@@ -0,0 +1,22 @@
+import java.util.Set;
+
+public class MinusFilter {
```

Comment:
Since all three types of filters have one similar method you could use an interface or an abstract class.

Reply:
That's right but these methods aren't the same because their inputs are different.